outcomes/firefox_desktop/spotlight_engagement.toml

```
@@ -1,24 +1,41 @@
-friendly_name = "Spotlight Engagement - not experiment specific"
+### outcome properties
+friendly_name = "Spotlight Engagement - not experiment specific"  # TODO: is the not experiment specific still accurate?
 description = "Usage and engagement metrics for the Spotlight feature - for ALL engagement with Spotlight feature, not experiment specific "
 
+
+### parameters definition
+[parameters]
+
+[parameters.message_id]
+friendly_name = "Message ID associated with the experiment"
+description = "ID of the spotlight message associated with this experiment"
+default = "'AND DIV(7, 0)' -- Failed to specify a value for message_id"
+
+[parameters.branch]
+friendly_name = "Branch name associated with the experiment"
+description = "Branch name associated with this experiment's message id"
+default = "'AND DIV(11, 0)' -- Failed to specify a value for branch_id"
+
+
+### metrics definition
 [metrics.spotlight_impressions]
 friendly_name = "Spotlight Impressions"
 description = "How often users saw Spotlight during an analysis window"
-select_expression = "COUNTIF(event = 'IMPRESSION')"
+select_expression = "COUNTIF(event = 'IMPRESSION' AND message_id = '{{ parameters.message_id }}' AND e.branch = '{{ parameters.branch }}')"
```

```suggestion
 select_expression = "COUNTIF(event = 'IMPRESSION' AND message_id = '{{ parameters.message_id }}'"
```

Jetstream will replace `message_id = '{{ parameters.message_id }}'` with something like (probably slightly different depending on the implementation):

```
message_id = '{{parameters.message_id.branch_1.value}}' AND e.branch = '{{parameters.branch_1.value}}') OR (message_id = '{{parameters.message_id.branch_2.value}}' AND e.branch = '{{parameters.branch_2.key}}') OR …)
```

to ensure we are looking at the right message ID for the correct branch. Instead of `branch_1`, it will use the actual branch name, those are just examples/placeholders here.